http/src/main/scala/plans/UI.scala

```diff
@@ -53,71 +57,71 @@ final case class UI(config: NelsonConfig) extends Default {
     case GET -> Root / "profile" & IsAuthenticated(s) =>
       Uri.fromString(s"/profile/${s.user.login}").fold(
         e => InternalServerError(s"Bad redirect: ${e.details}"),
-        Found.apply
+        uri => Found(Location(uri))
       )
 
     case GET -> Root / "profile" / owner & IsAuthenticated(_) =>
-      serveStaticFile(path = "index.html")
+      IO.pure(serveStaticFile(path = "index.html"))
 
     case GET -> Root / "deployments" & IsAuthenticated(_) =>
-      serveStaticFile(path = "index.html")
+      IO.pure(serveStaticFile(path = "index.html"))
 
     case GET -> Root / "deployments" / _ & IsAuthenticated(_) =>
-      serveStaticFile(path = "index.html")
+      IO.pure(serveStaticFile(path = "index.html"))
 
     case GET -> Root / "datacenters" & IsAuthenticated(_) =>
-      serveStaticFile(path = "index.html")
+      IO.pure(serveStaticFile(path = "index.html"))
 
     case GET -> Root / "datacenters" / _ & IsAuthenticated(_) =>
-      serveStaticFile(path = "index.html")
+      IO.pure(serveStaticFile(path = "index.html"))
 
     case GET -> Root / "dashboard" & IsAuthenticated(_) =>
-      serveStaticFile(path = "index.html")
+      IO.pure(serveStaticFile(path = "index.html"))
 
     case GET -> Root / "dashboard" / _ & IsAuthenticated(_) =>
-      serveStaticFile(path = "index.html")
+      IO.pure(serveStaticFile(path = "index.html"))
 
     case GET -> Root / "audit" / _ & IsAuthenticated(_) =>
-      serveStaticFile(path = "index.html")
+      IO.pure(serveStaticFile(path = "index.html"))
 
     case GET -> Root / "audit" & IsAuthenticated(_) =>
-      serveStaticFile(path = "index.html")
+      IO.pure(serveStaticFile(path = "index.html"))
 
     ////////////////////// REPOSITORIES //////////////////////
 
     case GET -> Root / owner / repo / unit ~ "svg" =>
       Nelson.getStatusOfReleaseUnit(Slug(owner, repo), unit)(config)
         .flatMap { s =>
-          Ok(badge(s.getOrElse(DeploymentStatus.Unknown)).toString)
-            .withContentType(Some(MediaType.`image/svg+xml`))
+          Ok(badge(s.getOrElse(DeploymentStatus.Unknown)).toString).map(
+            _.withContentType(`Content-Type`(MediaType.`image/svg+xml`)))
         }
 
     case NotAuthenticated() =>
-      Found(uri("/login"))
+      Found(Location(uri("/login")))
   }
 
   /*
    * if the server config has been overriden to provide an fs path
    * then we'll try to serve the file from that location, otherwise,
    * we'll use the built-in nelson-ui that comes from the classpath dependency.
    */
-  private def serveStaticFile(path: String) =
+  private def serveStaticFile(path: String): Response[IO] =
```

Comment:
This seems like an `IO[Response[IO]]` to me.

Reply:
Wdym? It type checks? :D

Reply:
I'm saying I think this should return an `IO[Response[IO]]` because it's, like, doing `IO`.  And then a bunch of those other changes you made melt away.

Reply:
Ohh I see, yeah you're right. 👍

Reply:
@rossabaker Hmm taking a closer look now I'm less sure. I assume by "doing `IO`" you are referring to `fileFromFilesystem` and `fileFromClasspath`, both of which give back `Response[IO]` - any `IO` being done in there is done through FS2 which gives a `fs2.Stream[IO, _]` and stuffed in a `Response` which gives `Response[IO]` not `IO[Response[IO]]`. Unless this is a bug with how HTTP4S let's you construct `Response`s?

Reply:
Oh, I see.  I thought you were using http4s' staticcontent helpers, which return an `OptionT[F, Response[F]]`.

So this method of constructing a response is constructing 200 OK responses that might fail.  You might want to do an `IO` to ensure that the file is there and readable and then return the stream, or else return a 404.  But what you've done is also legit.